### PR TITLE
Add a provider for linked collaboration resources

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Based on https://github.com/nextcloud/server/pull/14664
- [x] Add resource provider
- [x] Register resource provider
- [x] Clear access cache on participant changes
- [x] Add the UI

image
image
image

### CODE REVIEW

File: lib/Collaboration/Resources/ConversationProvider.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	public function __construct(Manager $manager) {
+		$this->manager = $manager;
+	}
+
+	public function getResourceRichObject(IResource $resource): array {
+		try {
+			$room = $this->manager->getRoomByToken($resource->getId());
+
+			return [
+				'type' => 'file',
```
Comment (juliushaertl) [inline lib/Collaboration/Resources/ConversationProvider.php]:
```suggestion
				'type' => 'room',
```

File: lib/Collaboration/Resources/ConversationProvider.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		}
+	}
+
+	public function canAccessResource(IResource $resource, IUser $user = null): bool {
+		try {
+			$room = $this->manager->getRoomForParticipantByToken(
+				$resource->getId(),
+				$user instanceof IUser ? $user->getUID() : null
+			);
+			return $room->getType() === Room::PUBLIC_CALL;
```
Comment (juliushaertl) [inline lib/Collaboration/Resources/ConversationProvider.php]: Shouldn't this also be true if the user is part of the call?

Comment (nickvergessen) [reply lib/Collaboration/Resources/ConversationProvider.php]: Right, I messed this up. Will fix

**Comment by juliushaertl** [comment]
Added some screenshots of the UI to the first post.

**Comment by juliushaertl** [comment]
@nickvergessen Frontend part is ready now.

**Comment by nickvergessen** [comment]
Bildschirmfoto von 2019-03-25 16-08-31

- [ ] Current room is listed normally in the UI to select a room to link to, should be disabled?
- [ ] There is a white box below the scrollbar in dark theme
---

- [ ] (+) button in the file picker is missing some CSS beauty (it's not round but has height 16px)
---

Only public rooms work atm

**Comment by nickvergessen** [comment]
- [ ] Icon is black on dark-theme. I guess they should all be delivered in black all the time (for android and other clients) and then be inverted to white in the collection UI?

**Comment by juliushaertl** [comment]
>  (+) button in the file picker is missing some CSS beauty (it's not round but has height 16px)

This will need a fix in the files app.

**Comment by juliushaertl** [comment]
>  Icon is black on dark-theme. I guess they should all be delivered in black all the time (for android and other clients) and then be inverted to white in the collection UI?

We have a class available depending on the resource type so we could probably set some individual icons for the web frontend and hide the regular image in that case. For talk this is `.resource-type-room`

**Comment by nickvergessen** [comment]
> We have a class available depending on the resource type so we could probably set some individual icons for the web frontend and hide the regular image in that case. For talk this is .resource-type-room

Or we do it similar to the navigation and invert the icons with a filter in the webUI?
Or well let's just take over the icons so we can render the room icon in the future.

**Comment by juliushaertl** [comment]
> Or we do it similar to the navigation and invert the icons with a filter in the webUI?
Or well let's just take over the icons so we can render the room icon in the future.

I think we shouldn't invert in general, as for the files app for example, we already use mimetype icons, which should not be inverted.

**Comment by nickvergessen** [comment]
Okay, ready for testing I guess

**Comment by nickvergessen** [comment]
Rebased merge conflicts after webrtc merge

**Comment by nickvergessen** [comment]
* Fixed highlight when room is selected 
* Excluded file/password rooms as they are temporary and added a note to #1616 to exclude that too

**Comment by nickvergessen** [comment]
Requires https://github.com/nextcloud/server/pull/14892 on the server.

**Pull request merged**